File: solutions/add-digits/add-digits.cpp
// @Title: Add Digits
// @Author: colinjxc
// @Date: 2022-01-26T19:52:24+08:00
// @URL: https://leetcode-cn.com/problems/add-digits


class Solution {
public:
    int addDigits(int num) {
        while (num > 9) {
            int res = 0;
            while (num > 0) {
                res += num%10;
                num /= 10;
            }
            num = res;
        }
        return num;
    }
};

File: solutions/add-digits/add-digits.go
// @Title: Add Digits
// @Author: colinjxc
// @Date: 2022-01-26T21:01:01+08:00
// @URL: https://leetcode-cn.com/problems/add-digits


func addDigits(num int) int {
    if num > 9 {
        return addDigits(func(n int) int {
            res := 0
            for n > 0 {
                res += n % 10
                n /= 10
            }
            return res
        }(num))
    }
    return num
}
